File: 백준/Silver/1026. 보물/보물.py

N = int(input())

A = [int(n) for n in input().split(' ')]
B = [int(n) for n in input().split(' ')]

A.sort(reverse=True)
B.sort()

result = 0
for i in range(N):
    result += A[i] * B[i]

print(result)

File: 백준/Silver/1026. 보물/보물.cc
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> a(n);
  vector<int> b(n);
  for (int i = 0; i < n; i++) {
    cin >> a[i];
  }
  for (int i = 0; i < n; i++) {
    cin >> b[i];
  }

  sort(a.begin(), a.end());
  sort(b.begin(), b.end(), greater<int>());

  int result = 0;
  for (int i = 0; i < n; i++) {
    result += a[i] * b[i];
  }

  cout << result << endl;

  return 0;
}